File: frontend/src/services/bookingService.js
import api from "./api";

export const createBooking = (bookingData) => {
    return api.post("/bookings", bookingData);
};

export const getMyBookings = () => {
    return api.get("/bookings/my");
};

export const getBookingById = (bookingId) => {
    return api.get(`/bookings/${bookingId}`);
};

export const cancelBooking = (bookingId) => {
    return api.delete(`/bookings/${bookingId}`);
};

File: backend/src/main/java/com/stayease/service/BookingService.java
package com.stayease.service;

import com.stayease.dto.request.BookingRequest;
import com.stayease.dto.response.BookingResponse;

import java.util.List;

public interface BookingService {

    BookingResponse bookRoom(String email, BookingRequest request);

    List<BookingResponse> getMyBookings(String email);

    BookingResponse getBookingById(Long bookingId);

    void cancelBooking(Long bookingId);
}